test/org/apache/catalina/core/TestStandardContextValve.java

```diff
@@ -182,4 +184,66 @@ public void requestDestroyed(ServletRequestEvent sre) {
         }
 
     }
+
+    @Test
+    public void test100ContinueDefaultPolicy() throws Exception {
+        // the default policy is IMMEDIATELY
+        // This test verifies that we get proper 100 Continue responses
+        // when the continueHandlingResponsePolicy property is not set
+        final Tomcat tomcat = getTomcatInstance();
+
+        final Connector connector = tomcat.getConnector();
+        connector.setProperty("continueHandlingResponsePolicy", "IMMEDIATELY");
+
+        test100Continue();
+    }
+
+    @Test
+    public void test100ContinueSentImmediately() throws Exception {
+        final Tomcat tomcat = getTomcatInstance();
+
+        final Connector connector = tomcat.getConnector();
+        connector.setProperty("continueHandlingResponsePolicy", "IMMEDIATELY");
+
+        test100Continue();
+    }
+
+    @Test
+    public void test100ContinueSentOnRequestContentRead() throws Exception {
+        final Tomcat tomcat = getTomcatInstance();
+
+        final Connector connector = tomcat.getConnector();
+        connector.setProperty("continueHandlingResponsePolicy", "ON_REQUEST_CONTENT_READ");
+
+        test100Continue();
+    }
+
+    public void test100Continue() throws Exception {
```

Comment:
I am bit confused by the test. How does it verify that the new behavior works if the test is identical for both cases?

Reply:
I struggled a bit with writing the tests, in both cases a 100 continue response is expected. I'll give testing some more thought.

Reply:
I've reworked the test, please let me know what you think of if you have other ideas on how to test this change.